### PR TITLE
[Issue#566]Search Bar auto-complete

### PR DESCRIPTION
**Pull request by AC-Unicorn**
1. add proxy script, and bash script under script folder for starting the proxy
2.In main.scala included latest version of jquery-ui.js and its css file
3.modified searchbar/controllers.js to implement auto-complete feature
4.Changed style file stylesheet/main.css for polishing auto-complete menu

### CODE REVIEW

**Comment by chenlica** [comment]
@AC-Unicorn Quick comment: don't use names in the PR subject.  Make it more general and informative.

File: examples/twittermap/web/app/views/twittermap/main.scala.html [hunk 1]
```diff
@@ -69,7 +70,7 @@
   <script src='@routes.Assets.versioned("lib/dc.js/dc.min.js")'></script>
   <script src='@routes.Assets.versioned("lib/bootstrap/js/bootstrap.min.js")'></script>
   <script src='@routes.Assets.versioned("lib/bootstrap-toggle/js/bootstrap-toggle.min.js")'></script>
-  <script src='@routes.Assets.versioned("lib/jquery-ui/jquery-ui.js")'></script>
+  <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
```
Comment (lengyyy) [inline examples/twittermap/web/app/views/twittermap/main.scala.html]: It will be better to use the offline js file.

Comment (lengyyy) [reply examples/twittermap/web/app/views/twittermap/main.scala.html]: Refer to #540, there is the method about how to add a third party library.

File: examples/twittermap/web/public/javascripts/searchbar/controllers.js [hunk 2]
```diff
@@ -1,7 +1,36 @@
 angular.module('cloudberry.util', ['cloudberry.common'])
   .controller('SearchCtrl', function($scope, $window, cloudberry, cloudberryConfig, moduleManager) {
       var stopwordsMap = buildStopwordsMap();
+        
+        $("#keyword-textbox").autocomplete({source:[]});
```
Comment (baiqiushi) [inline examples/twittermap/web/public/javascripts/searchbar/controllers.js]: Let's make the indentation format consistent with current code.

File: examples/twittermap/web/public/javascripts/searchbar/controllers.js [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+        $("#keyword-textbox").on("keyup",function(event){
+          
+          var q = $scope.keyword;
+          var url = "http://127.0.0.1:5000/spoof?query="+q;
```
Comment (baiqiushi) [inline examples/twittermap/web/public/javascripts/searchbar/controllers.js]: The `proxy` server should be running on our `Twittermap` server side, then this `127.0.0.1` address will not work. There're 2 options to implement this: (1) make your `ajax` request to a relative path with the same domain as this website is from; (2) make this absolute address configurable in our `application.conf` of `Twittermap` application.

File: examples/twittermap/web/public/javascripts/searchbar/controllers.js [hunk 4]
```diff
@@ -1,7 +1,36 @@
...
+          
+            $.ajax({url:url}).done(function(data){
```
Comment (baiqiushi) [inline examples/twittermap/web/public/javascripts/searchbar/controllers.js]: Same, make indentation correct.

**Comment by baiqiushi** [comment]
Also, we need to consider what will happen in the frontend if the `proxy` server is down. The purpose should be that all frontend features except auto-complete will not be influenced if the `proxy` server is down.

File: examples/twittermap/web/public/javascripts/searchbar/controllers.js [hunk 5]
```diff
@@ -1,7 +1,39 @@
 angular.module('cloudberry.util', ['cloudberry.common'])
   .controller('SearchCtrl', function($scope, $window, cloudberry, cloudberryConfig, moduleManager) {
       var stopwordsMap = buildStopwordsMap();
+        
+      $("#keyword-textbox").autocomplete({source:[]});
```
Comment (baiqiushi) [inline examples/twittermap/web/public/javascripts/searchbar/controllers.js]: Let's add a brief comment for this block of code like you did for line 28.

File: examples/twittermap/web/public/javascripts/searchbar/controllers.js [hunk 6]
```diff
@@ -1,7 +1,39 @@
...
+                  suggestion.push(value);
+                }
+              }
+              $("#keyword-textbox").autocomplete({source:suggestion});
+          });
+        }
+        catch(err){}
+      })
+        
+      //If keyword been selected and user pushed enter,then we perform search directly
```
Comment (baiqiushi) [inline examples/twittermap/web/public/javascripts/searchbar/controllers.js]: The indentation format of this block is not correct.

**Comment by chenlica** [comment]
Superb!

**Pull request merged**